src/monmove.cpp

```diff
@@ -418,16 +418,42 @@ void monster::move()
     int mondex = !plans.empty() ? g->mon_at( plans[0], is_hallucination() ) : -1;
     auto mon_att = mondex != -1 ? attitude_to( g->zombie( mondex ) ) : A_HOSTILE;
 
-    if( !plans.empty() &&
-        ( mon_att == A_HOSTILE || has_flag( MF_ATTACKMON ) || has_flag( MF_PUSH_MON ) ) &&
-        ( can_move_to( plans[0] ) ||
-          ( plans[0] == g->u.pos3() ) ||
-          ( ( has_flag( MF_BASHES ) || has_flag( MF_BORES ) ) &&
-            g->m.bash_rating( bash_estimate(), plans[0] ) >= 0 ) ) ) {
-        // CONCRETE PLANS - Most likely based on sight
-        next = plans[0];
-        moved = true;
-    } else if( has_flag( MF_SMELLS ) ) {
+    // CONCRETE PLANS - Most likely based on sight
+    if( !plans.empty() ) {
+        // When attacking an adjacent enemy, we're direct.
+        if( (mondex != -1 && mon_att == A_HOSTILE) || plans[0] == g->u.pos3() ) {
```

Comment:
That's a good point, how do they end up attacking NPCs? or do they?  ISTR reports of them just standing next to NPCs and getting killed.